HW#4 - Nakhimov Ilya

nakhimov-i opened the pull request.

cafecco commented on script.js:
```diff
@@ -0,0 +1,178 @@
+Order.prototype.pay = function(){
+    this.isPaid = true;
+    console.log("Thank you! Your order has been paid successfully.")
+}
```
If it's not difficult to you, could you please create instances of items and actions with the order? It's OK, if you don't, because it wasn't written in the task description, and I will do it myself just to check all the program:)

nakhimov-i commented:
@cafecco 

Alena, thank you for the code review!

I fixed the bugs:
    1. Got rid of magic numbers.
    2. Removed duplicate methods.
    3. Removed unnecessary parts of the code.

Also, I added examples of actions for comfortable code review)

nakhimov-i commented on script.js:
```diff
@@ -0,0 +1,178 @@
+    MenuItem.call(this, size.price + stuffing.price, size.calories + stuffing.calories);  
+}
+
+Hamburger.prototype = Object.create(MenuItem.prototype);
```
![hw4](https://user-images.githubusercontent.com/51382430/73845150-115c1200-4833-11ea-9810-42783bdae788.png)

I didn't really understand this point.
If we open the Hamburger object, all the constructor properties we need will be there.
Or did you mean something else?

cafecco replied:
You're right, the properties of the certain object will be those which we expect. But you lose child's constructor without its explicit indication. Look, if you want to see `Hamburger.prototype.constructor` you'll get 
![image](https://user-images.githubusercontent.com/29711636/73922003-74a07f80-48d9-11ea-8c5b-42dc0f0cb950.png)
but regarding inheritance rules it's important to save child's constructor with this way:
`Hamburger.prototype.constructor = Hamburger;`
and you should do that strictly **after** this line:
`Hamburger.prototype = Object.create(MenuItem.prototype);`
(if it's not clear why it's used in this order, write me in Slack and I'll tell you it in details).
So, after that you can check child's constructor: 
![image](https://user-images.githubusercontent.com/29711636/73922567-6dc63c80-48da-11ea-9d59-3d942ade9d13.png)

cafecco commented on script.js:
```diff
@@ -0,0 +1,191 @@
+//salad
+
+function Salad (type, weight){
+    MenuItem.call(this, type.price * weight / 100, type.calories * weight / 100 );
```
I still see magic number :)
Could you please create variable in Salad constructor, for example:
`var basicWeight = 100;` 
and replace with it where it's needed?

nakhimov-i replied:
> I still see magic number :)
> Could you please create variable in Salad constructor, for example:
> `var basicWeight = 100;`
> and replace with it where it's needed?

My bad. I fixed that!
